### PR TITLE
fix: restricting executor role to Governor

### PR DESCRIPTION
**Pull request by wei3erHase**

### CODE REVIEW

File: src/contracts/governance/HaiGovernor.sol [hunk 1]
```diff
@@ -45,8 +45,7 @@ contract HaiGovernor is
     TimelockController _timelock = TimelockController(payable(timelock()));
     _timelock.grantRole(keccak256('PROPOSER_ROLE'), address(this));
     _timelock.grantRole(keccak256('CANCELLER_ROLE'), address(this));
-
-    _timelock.grantRole(keccak256('EXECUTOR_ROLE'), address(0));
+    _timelock.grantRole(keccak256('EXECUTOR_ROLE'), address(this));
```
Comment (wei3erHase) [inline src/contracts/governance/HaiGovernor.sol:48]: main change, setting it to address 0 is making the role open, now txs have to be executed through governor

File: test/e2e/E2EGovernor.t.sol [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        IGovernor.GovernorUnexpectedProposalState.selector, _proposalId, IGovernor.ProposalState.Active, 0x10
+      )
+    );
+    haiGovernor.queue(targets, values, callDatas, keccak256(bytes(description)));
+
+    vm.warp(block.timestamp + _governorParams.votingPeriod + 1);
+    haiGovernor.queue(targets, values, callDatas, keccak256(bytes(description)));
+
+    vm.expectRevert(
+      // abi.encodeWithSelector(TimelockController.TimelockUnexpectedOperationState.selector, PROPOSAL_HASH, 0x2)
```
Comment (wei3erHase) [inline test/e2e/E2EGovernor.t.sol:51]: cannot get the hash ⚰️

File: script/Contracts.s.sol [hunk 3]
```diff
@@ -91,7 +91,6 @@ import {HaiDelegatee} from '@contracts/governance/HaiDelegatee.sol';
  */
 abstract contract Contracts {
   // --- Helpers ---
-  uint256 public chainId;
```
Comment (wei3erHase) [inline script/Contracts.s.sol:94]: as requested in https://github.com/hai-on-op/core/pull/32

File: test/e2e/E2EGovernor.t.sol [hunk 4]
```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.20;
+
+import {HaiTest} from '@test/utils/HaiTest.t.sol';
+import {Deploy, DeployMainnet, DeployMainnet} from '@script/Deploy.s.sol';
```
Comment (0xJabberwock) [inline test/e2e/E2EGovernor.t.sol]: `DeployMainnet` shows twice in the statement.

File: test/e2e/E2EGovernor.t.sol [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+  function setUp() public override {
+    vm.createSelectFork(vm.rpcUrl('mainnet'), FORK_BLOCK);
+    super.setUp();
+    run();
+
+    vm.prank(address(timelock));
+    protocolToken.mint(whale, 1_000_000e18);
+    vm.prank(whale);
+    protocolToken.delegate(whale);
+    vm.warp(block.timestamp + 1);
```
Comment (0xJabberwock) [inline test/e2e/E2EGovernor.t.sol:96]: Seems that `E2EGovernorTest` always requires this setup. What about moving it within its scope?

File: test/e2e/E2EGovernor.t.sol [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+    targets[0] = address(protocolToken);
+    callDatas[0] = abi.encodeWithSignature('unpause()');
+    values[0] = 0;
+    description = 'Unpause the protocol';
+
+    vm.startPrank(whale);
+    uint256 _proposalId = haiGovernor.propose(targets, values, callDatas, description);
+
+    vm.expectRevert();
+    protocolToken.transfer(address(0x69), 1);
```
Comment (0xJabberwock) [inline test/e2e/E2EGovernor.t.sol:28]: I suggest replacing `address(0x69)` with `random`.

File: test/e2e/E2EGovernor.t.sol (hunk 5 above)
Comment (wei3erHase) [reply test/e2e/E2EGovernor.t.sol:96]: hmm don't have a `setup` in that scope :/

File: test/e2e/E2EGovernor.t.sol (hunk 4 above)
Comment (wei3erHase) [reply test/e2e/E2EGovernor.t.sol]: lol, didn't know is possible

File: test/e2e/E2EGovernor.t.sol (hunk 6 above)
Comment (0xJabberwock) [reply test/e2e/E2EGovernor.t.sol:28]: The highlighted `address(0x69)` wasn't replaced. LUL

File: test/e2e/E2EGovernor.t.sol (hunk 5 above)
Comment (0xJabberwock) [reply test/e2e/E2EGovernor.t.sol:96]: Why not creating it?

**Pull request merged**